internal/processors/cloudeventconvert/cloudeventconvert.go

```diff
@@ -94,37 +96,137 @@ func (c *cloudeventProcessor) processMsg(ctx context.Context, msg *service.Messa
+			data, marshalErr := json.Marshal(err)
+			if marshalErr == nil {
+				msg.SetBytes(data)
+			}
+			processors.SetError(msg, processorName, "failed to convert to cloud event", err)
+			return service.MessageBatch{msg}
+		}
+		if len(hdrs) == 0 {
+			processors.SetError(msg, processorName, "no cloud events headers returned", nil)
+			return service.MessageBatch{msg}
+		}
+		if len(eventData) == 0 {
+			// If the module chooses not to return data, use the original message will be used
+			eventData = msgBytes
+		}
+	case httpinputserver.AttestationContent:
+		var event cloudevent.CloudEvent[json.RawMessage]
+		err = json.Unmarshal(msgBytes, &event)
+		if err != nil {
+			processors.SetError(msg, processorName, "failed to unmarshal attestation cloud event", err)
+			return service.MessageBatch{msg}
+		}
+
+		currentTimestamp := time.Now().UTC()
+		if !(currentTimestamp.Sub(event.Time).Minutes() <= 5 && event.Time.Sub(currentTimestamp).Minutes() <= 5) {
+			processors.SetError(msg, processorName, fmt.Sprintf("event timestamp %+v exceeds valid range", event.Time), nil)
+			return service.MessageBatch{msg}
+		}
+
+		if event.ID == "" {
+			event.ID = ksuid.New().String()
+		}
+
+		if event.DataContentType == "" {
+			event.DataContentType = "application/json"
+		}
+
+		if event.SpecVersion == "" {
+			event.SpecVersion = "1.0"
+		}
+
+		if event.Type == "" {
+			event.Type = "dimo.attestation"
+		}
+
+		if _, err := cloudevent.DecodeNFTDID(event.Subject); err != nil {
+			processors.SetError(msg, processorName, "invalid attestation subject format", err)
+			return service.MessageBatch{msg}
+		}
+
+		validSignature, err := c.validateSignature(msg, event.Producer)
+		if err != nil {
+			processors.SetError(msg, processorName, "failed to validate signature on message", err)
+			return service.MessageBatch{msg}
+		}
+
+		if !validSignature {
+			processors.SetError(msg, processorName, "message signature invalid", err)
+			return service.MessageBatch{msg}
+		}
+
+		hdrs = append(hdrs, event.CloudEventHeader)
+		msg.MetaSetMut(CloudEventIndexValueKey, hdrs)
 	}
-	retBatch, err := c.createEventMsgs(msg, source, hdrs, eventData)
+
+	var retBatch []*service.Message
+	retBatch, err = c.createEventMsgs(msg, source, hdrs, eventData)
 	if err != nil {
 		processors.SetError(msg, processorName, "failed to create event messages", err)
 		return service.MessageBatch{msg}
 	}
 	return retBatch
 }
 
+func (c *cloudeventProcessor) validateSignature(msg *service.Message, producer string) (bool, error) {
+	producerDID, err := cloudevent.DecodeEthrDID(producer)
+	if err != nil {
+		return false, errors.New("failed to decode producer did")
+	}
+
+	msgBytes, err := msg.AsBytes()
+	if err != nil {
+		return false, errors.New("failed to parse message as bytes")
+	}
+
+	var event cloudevent.CloudEvent[json.RawMessage]
+	err = json.Unmarshal(msgBytes, &event)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse cloud event: %w", err)
+	}
```

Comment:
We already unmarshal the event before this function is called can we just use that